src/renderer/vsync/vsync_win.rs

```diff
@@ -1,64 +1,115 @@
 use std::{
     sync::{
         atomic::{AtomicBool, Ordering},
-        Arc, Condvar, Mutex,
+        Arc,
     },
     thread::{spawn, JoinHandle},
+    time::Duration,
 };
-use winapi::um::dwmapi::DwmFlush;
+use winapi::{
+    shared::ntdef::LARGE_INTEGER,
+    um::dwmapi::{DwmGetCompositionTimingInfo, DWM_TIMING_INFO},
+    um::profileapi::{QueryPerformanceCounter, QueryPerformanceFrequency},
+};
+
+use winit::event_loop::EventLoopProxy;
+
+use spin_sleep::SpinSleeper;
 
-use crate::profiling::tracy_zone;
+use crate::{
+    profiling::{tracy_plot, tracy_zone},
+    window::UserEvent,
+};
 
 pub struct VSyncWin {
     should_exit: Arc<AtomicBool>,
     vsync_thread: Option<JoinHandle<()>>,
-    vsync_count: Arc<(Mutex<usize>, Condvar)>,
-    last_vsync: usize,
+    redraw_requested: Arc<AtomicBool>,
+}
+
+/// Calculates the time until the vblank, taking into account that the vblank is cyclic, so this
+/// always finds the next vblank forward
+fn time_until_vblank_forward(delay: f64, period: f64) -> f64 {
+    delay.rem_euclid(period)
+}
+
+/// Calculates the time to wait to hit the vblank + offset given a period Note always tries to wait
```

Comment:
`tq` probably had fun here.

```suggestion
/// Calculates the time to wait to hit the vblank + offset given a period
///
/// Note: always tries to wait
```